src/boot/efi/efi-string.c

```diff
@@ -173,6 +182,37 @@ char16_t *strchr16(const char16_t *s, char16_t c) {
         STRCHR_U(char16_t *);
 }
 
+#define STRNDUP_U(type, _len)                             \
+        ({                                                \
+                if (!s)                                   \
+                        return NULL;                      \
+                                                          \
+                size_t len = _len;                        \
+                size_t size = len * sizeof(type);         \
+                                                          \
+                type *dup = xmalloc(size + sizeof(type)); \
+                efi_memcpy(dup, s, size);                 \
+                dup[len] = '\0';                          \
```

Comment:
Maybe `u'\0'` for `char16_t`??

Reply:
Doesn't matter because it gets promoted to `char16_t` (boot.c does basically the same for long char literals). Also, this needs to work for both `char` and `char16_t`.

Reply:
And now I've side-stepped the matter by using `memset`. That way I can use `void*` and drop the `type` arg.